specification/transport/abstract.tex

```diff
@@ -662,16 +662,19 @@ \subsubsection{Transmission over redundant transports}
     in section~\ref{sec:application_functions}.
 }.
 
-\subsection{Transfer reception}
+\subsection{Transfer reception}\label{sec:transport_transfer_reception}
 
 \subsubsection{Definitions}
 
 \emph{Transfer reassembly} is the real-time process of reconstruction of the transfer payload and its metadata from
 a sequence of relevant transport frames.
 
-\emph{Transfer-ID timeout} is a time interval that is 2 (two) seconds long.
-The semantics of this entity are explained below.
-Implementations are allowed to redefine this value provided that such redefinition is explicitly documented.
+\emph{Transfer-ID timeout} is a time interval whose semantics are explained below.
+Implementations may define this value statically according to the application requirements.
+Implementations may automatically adjust this value per session at runtime as a function of the
+observed transfer reception interval.
+Transfer-ID timeout values greater than 2 (two) seconds are not recommended.
```

Comment:
where do we specify what to do if a node gets a transfer using the same session identifier but with a new transfer id? My assumption is this should reset the pending session?

Reply:
Yes. In v0 we used to be extremely hands-on about the reception logic, but at one point I realized that it is too limiting because implementations should be able to choose specific approaches as long as the *externally observable behavior* is spec-compliant. In v1 we only state the high-level requirements and constraints, allowing implementers to decide how to meet them. To your question specifically, we say this:

**4.1.4.1 Definitions:** An ordered transfer sequence is a sequence of transfers whose temporal order is covariant with their transfer-ID values.

**4.1.4.2 Behaviors:** Reassembled transfers shall form an ordered transfer sequence.

Ergo, if a new transfer is started while the previous one is not yet completed, the node should discard the old transfer because not doing so in the hopes that it will be eventually completed after the new one is finished may violate the ordering requirement.

The specification is not a legal document though, and it is open to malicious interpretation. One could choose to do the following while being technically spec-conforming:

- Delay the reception of the new transfer indefinitely until the old one is completed.
- Drop the new transfer in favor of completing the old one.

I am not sure if we should bother explicitly banning such behaviors.

Reply:
Dropping the old transfer and starting the new seems like the best light-weight behaviour. The other valid behaviour is to support multiple, interleaved transfers as long as all the transfer ID timeout never expires while each is being assembled. Of course this second, complex implementation only makes sense if a transmitting node has logic that may result in interleaved transfers (I suppose this becomes more plausible for nodes running a multi-tasking operating system that uses concurrent queues between the applications and the UAVCAN media layer). 

This is just me thinking this over. Not really a PR comment. I feel like I remember discussing this in the past but it's been awhile, sorry.